include/gadget/filter.h

```diff
@@ -119,5 +123,24 @@ static __always_inline bool gadget_should_discard_data_current()
 
 	return false;
 }
+#endif
+
+// gadget_should_discard_data_by_skb returns true if the gadget should skip this
+// event based on the fields of the skb_val struct.
+static __always_inline bool
+gadget_should_discard_data_by_skb(struct sockets_value *skb_val)
+{
+	if (!skb_val) {
+		// drop the packet if any filtering is specified
+		if (gadget_filter_by_mntns || targ_pid != 0 || targ_tid != 0 ||
+		    targ_uid != GADGET_INVALID_ID ||
+		    targ_gid != GADGET_INVALID_ID || targ_comm[0] != '\0')
+			return true;
+	}
+
+	return gadget_should_discard_data(
+		skb_val->mntns, skb_val->pid_tgid >> 32, skb_val->pid_tgid,
+		skb_val->task, skb_val->uid_gid, skb_val->uid_gid >> 32);
```

Comment:
Comment not for this PR:
We should maybe create another function where we could just give `pid_tgid` and `uid_gid`. These shifts are easy to mess up imo

Reply:
Perhaps create some helpers to extract those fields is better? I suppose there is not any overhead on doing something like `gadget_should_discard_data(..., gadget_get_pid(pid_tdig), gadget_get_tid(pid_tgid)`, probably they can be just macros too.